woocommerce/woocommerce
# Fix block templates not being rendered in extension taxonomies

### Changes proposed in this Pull Request:

Alternative approach from #44646.

Closes https://github.com/woocommerce/woocommerce/issues/44453. (**Update:** it turns out https://github.com/woocommerce/woocommerce/pull/44537 already fixed this issue)

In some circumstances, `woocommerce_has_block_template` wasn't set to `true` correctly, that meant that we were rendering a broken PHP template instead of the correct block template.

The issue was only reproducible in block themes with certain custom WooCommerce templates (ie: Tsubaki, which has the `archive-product.html` template), and were only reproducible in these cases:

* When rendering a product archive page belonging to a taxonomy with no template in the theme. That's excluding the category, tag and attribute taxonomies. So it was only reproducible in taxonomies added by extensions.
* ~When rendering the Shop page with certain filters (what's described in #44453).~ (**Update:** This was already fixed in `trunk` thanks to https://github.com/woocommerce/woocommerce/issues/44453)

This PR removes a check we had in the code that would only set `woocommerce_has_block_template` if the template was not in the theme.

### How to test the changes in this Pull Request:

**Verify correct template is rendered with certain settings (#44453)**

(**Update**: this issue was actually fixed in https://github.com/woocommerce/woocommerce/pull/44537, but I'm adding the testing steps here just to verify there are no regressions)

1. With a theme with custom WooCommerce templates (ie: Tsubaki).
2. Go to WooCommerce > Settings > Products > Advanced and disable _Use the product attributes lookup table for catalog filtering_.
3. With a block theme with custom WooCommerce templates (ie: Tsubaki) go to `/shop/?product_cat=Clothing&filter_size=large`. Note: the values of `product_cat` and `filter_xyz` might vary depending on the categories and attributes of your store. In case of doubt, you can import the [products from `sample-data`](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce/sample-data).
4. Verify the page is rendered correctly. Ie: verify the header is displayed, verify the page has the _Shop_ title, verify the styles are loaded, etc.

Before | After
--- | ---
![image](https://github.com/woocommerce/woocommerce/assets/3616980/12413fda-2c24-49d7-8b21-3cbb4537776f) | ![image](https://github.com/woocommerce/woocommerce/assets/3616980/2ab71af9-754c-46db-be31-9e25598f0167)

**Verify taxonomies added by extensions are rendered correctly (testing for devs)**

1. With a theme with custom WooCommerce templates (ie: Tsubaki).
2. Install WooCommerce Brands.
3. In the admin, create a brand, edit a product and assign that product to the brand you just created.
4. Visit the brand page in the frontend (the page should render correctly).
5. Edit `class-wc-brands-block-templates.php` and comment out this line: `add_filter( 'woocommerce_has_block_template', array( $this, 'has_block_template' ), 10, 2 );`.
6. Visit the brand page again. Verify the page still renders correctly.

### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
